Pidaq/Lib/wind_tunnel/air_properties.py

```diff
@@ -0,0 +1,111 @@
+"""Contains class that defines linear interpolation functions for wind tunnel testing
+
+Classes:
+    LinearInterpolation
+"""
+
+import csv
+import logging
+import sys
+from scipy import interpolate
+
+class LinearInterpolation:
+    """Creates linear interpolation functions for density and dynamic viscosity
+
+    This class defines two functions that are used to linearly interpolate
+    density (kg/m3) and dynamic viscosity (Pa*S) based on temperature (°C)
+    for the purpose of wind tunnel testing.
+    
+    Attributes:
+        densityFunc
+        viscosityFunc
+
+    Methods:
+        interpolateDensity
+        interpolateViscosity
+    """
+
+    def __init__(self, path):
+        """
+        Reads a CSV file to define two functions for each of density and dynamic
+        viscosity that can be used with temperature.  The CSV should have same
+        structure as current one in Drive, where the units are in the second row.
+        
+        Parameters:
+            path - path to a CSV file of the air properties
+        """
+        self.logger = logging.getLogger()
+
+        try:
+            open(path) # Open the CSV file
+        except IOError as err:
+            self.logger.error(err)
+            sys.exit()
+        with open(path) as file:
+            csv_reader= csv.reader(file, delimiter =',')
+            next(csv_reader) # Skip the first row
+            tempCol, densityCol, viscosityCol = -1,-1,-1
+            row = next(csv_reader) # Read the second row that contains the units
+        
+            count = 0 # For tracking the column index
+            for cell in row: # This loop determines the column index for 
+                    # temperature, density, and dynamic viscosity based on units
+
+                if cell.find('°C') != -1:
+                    tempCol = count
+
+                if cell.find('[kg/m3]') != -1:
+                    densityCol = count
+
+                if cell.find('[Pa*S]') != -1:
+                    viscosityCol = count
+                count += 1
+
+            # if a unit is not found, terminate the script
+            if tempCol == -1: 
+                self.logger.error("Could not find \"°C\" in any column header.")
+                sys.exit()
```

Comment:
Not sure if this is the behavior that we want... think about it like this, this will close the whole application (web gui and controls), as it is being called from within that application, and not as a standalone script. We can discuss this

Reply:
Okay, makes sense. I believe you or Colton suggested I add another method that checks whether the units were found. What is the behavior that should be defined in that method and its return? If the units are not found, should I log an error and try to stop only this script without killing the whole application? Does raising an error work here? @Burke-Daniel @colton-smith

Reply:
I would say the most we would do is raise an error, and the least we would do is return an error code to be handled by the host application. For now, you can go ahead and raise an error.

As this is called during setup, I don't really see too much of a problem raising an error. I dont think it would close the application either, I think it would get logged by the dash application. If we dont like the behaviour, we can always change it later during testing.

Speaking of testing, there should really be a due diligence form that notes all the tests we do and the outcome for the entire wind tunnel application (just an idea)

Reply:
Alright, all calls to sys.exit() are removed and the appropriate errors are raised instead.